### PR TITLE
refactor(payment): STRIPE-423 created useHostedCreditCard hook

### PR DESCRIPTION
**Pull request by bc-dronov**
## What?
Created useHostedCreditCard hook.

## Why?
Because we have duplicated code in some payment providers, so it could be useful to avoid code duplication

## Testing / Proof
Manual tests
Example of usage is here: https://github.com/bigcommerce/checkout-js/pull/1964

@bigcommerce/team-checkout

### CODE REVIEW

File: packages/hosted-credit-card-integration/src/utils/index.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { useHostedCreditCard } from './useHostedCreditCard';
+
+export { useHostedCreditCard };
```
Comment (bc-peng) [inline packages/hosted-credit-card-integration/src/utils/index.ts:3]:
```suggestion
export { useHostedCreditCard } from './useHostedCreditCard';
```

File: packages/hosted-credit-card-integration/src/utils/useHostedCreditCard.spec.tsx
Comment (bc-peng) [inline packages/hosted-credit-card-integration/src/utils/useHostedCreditCard.spec.tsx:1]: We're getting rid of `enzyme`, could you use RTL for testing?

File: packages/hosted-credit-card-integration/src/utils/useHostedCreditCard.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    checkoutState,
+    method,
+    language,
+    paymentForm,
+}: UseHostedFormOptions) => {
+    const { setFieldTouched, setFieldValue, setSubmitted, submitForm } = paymentForm;
+    const {
+        config: { cardCode },
+    } = method;
+    const isCardCodeRequired = cardCode || cardCode === null;
```
Comment (bc-peng) [inline packages/hosted-credit-card-integration/src/utils/useHostedCreditCard.tsx:38]: Is there any other way to tell if cardCode is quired?
Depending on its own value seems not logical.

File: packages/hosted-credit-card-integration/src/utils/useHostedCreditCard.spec.tsx
Comment (bc-dronov) [reply packages/hosted-credit-card-integration/src/utils/useHostedCreditCard.spec.tsx:1]: Done, I've installed `@testing-library/react-hooks` for custom hooks testing.

File: packages/hosted-credit-card-integration/src/utils/useHostedCreditCard.tsx (hunk 2 above)
Comment (bc-dronov) [reply packages/hosted-credit-card-integration/src/utils/useHostedCreditCard.tsx]: As far as I know we only can depend on this variable from config, the same logic we have in `HostedCreditCardPaymentMethod.tsx`

File: packages/hosted-credit-card-integration/src/utils/index.ts (hunk 1 above)
Comment (bc-dronov) [reply packages/hosted-credit-card-integration/src/utils/index.ts]: Done

File: packages/hosted-credit-card-integration/src/index.ts [hunk 3]
```diff
@@ -2,3 +2,4 @@ export { default as HostedCreditCardPaymentMethod } from './HostedCreditCardPaym
 export { HostedCreditCardFieldsetValues, HostedCreditCardValidationValues } from './types';
 export { HostedCreditCardValidation } from './components/HostedCreditCardValidation';
 export { getHostedInstrumentValidationSchema } from './components/getHostedInstrumentValidationSchema';
+export * from './utils';
```
Comment (bc-peng) [inline packages/hosted-credit-card-integration/src/index.ts:5]: Could you please avoid using `*` here?

File: packages/hosted-credit-card-integration/src/utils/index.ts
Comment (bc-peng) [inline packages/hosted-credit-card-integration/src/utils/index.ts:1]: `utils` folder name seems not accurate.
Perhpas we can name it `hooks`.

File: packages/hosted-credit-card-integration/src/utils/useHostedCreditCard.tsx [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    const getHostedFormOptions = useCallback(
+        async (selectedInstrument: CardInstrument): Promise<HostedFormOptions> => {
+            const styleProps = ['color', 'fontFamily', 'fontSize', 'fontWeight'];
+            const isInstrumentCardNumberRequired = selectedInstrument
+                ? isInstrumentCardNumberRequiredProp(selectedInstrument, method)
+                : false;
+            const isInstrumentCardCodeRequired = selectedInstrument
+                ? isInstrumentCardCodeRequiredProp(selectedInstrument, method)
+                : false;
+            let styleContainerId;
```
Comment (bc-peng) [inline packages/hosted-credit-card-integration/src/utils/useHostedCreditCard.tsx:94]:
```suggestion
            let styleContainerId: string;
```

Comment (bc-dronov) [reply packages/hosted-credit-card-integration/src/utils/useHostedCreditCard.tsx]: Done

File: packages/hosted-credit-card-integration/src/utils/index.ts
Comment (bc-dronov) [reply packages/hosted-credit-card-integration/src/utils/index.ts:1]: Done

**Review by bc-peng — APPROVED** [review]
LGTM👏👏

**Pull request merged**